quarkiverse's LambdaBuildTimeConfig interface conflicts with io.quarkus's LambdaBuildTimeConfig class

Hello,

I'm trying to use `quarkus-amazon-lambda` from `io.quarkiverse .amazonservices` to inject AWS lambda client in a service that uses

```
       <dependency>
            <!-- AWS Lambda Client injection -->
            <groupId>io.quarkiverse.amazonservices</groupId>
            <artifactId>quarkus-amazon-lambda</artifactId>
        </dependency>

        <dependency>
             <!-- My service or some async job lambda -->
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-amazon-lambda-rest</artifactId> <!-- or just quarkus-lambda -->
       <dependency>
```



I tried all available versions of quarkus and quarkiverse and theirs combinations. However it doesn't work. You can try with the version combinations mentioned in https://github.com/quarkiverse/quarkus-amazon-services.

As a result my build fails with

> [ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.8.1:build (default) on project inventory-management-rest: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.amazon.lambda.deployment.LambdaProcessor#setupClient threw an exception: java.lang.IncompatibleClassChangeError: Found class io.quarkus.amazon.lambda.runtime.LambdaBuildTimeConfig, but interface was expected
[ERROR]         at io.quarkus.amazon.lambda.deployment.LambdaProcessor.setupClient(LambdaProcessor.java:104)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:840)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
[ERROR] -> [Help 1]


In [this commit](https://github.com/quarkiverse/quarkus-amazon-services/commit/9377e50a406db11355bf12b32e2bb09aee50f7c8#diff-78bd91a455473a18ad0cf5d49a53628328b9d028aa1338700415629815b3b310) you added `LambdaBuildTimeConfig` **interface** that conflicts with the `LambdaBuildTimeConfig` **class** used by io.quarkus. See [io quarkus class LambdaBuildTimeConfig](https://github.com/quarkusio/quarkus/blob/86ba00e29a9a326afa02a74dd81a8228a5d04c02/extensions/amazon-lambda/runtime/src/main/java/io/quarkus/amazon/lambda/runtime/LambdaBuildTimeConfig.java)

I guess the config interface package or the interface name need to be modified. So far a workaround is to not use quarkiverse and inject clients manually.

I'm have the need to write a lambda to invoke another lambda, so I though that using the `io.quarkiverse.amazonservices:quarkus-amazon-services` LambdaClient would be the correct approach.


On 2.8.1 I get the error mentioned in this issue

```
java.lang.RuntimeException: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.amazon.lambda.deployment.LambdaProcessor#discoverTelemetry threw an exception: java.lang.IncompatibleClassChangeError: Found class io.quarkus.amazon.lambda.runtime.LambdaBuildTimeConfig, but interface was expected
	at io.quarkus.amazon.lambda.deployment.LambdaProcessor.discoverTelemetry(LambdaProcessor.java:99)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
```

On 3.0.0.alpha1 the error below since I'm using https://quarkus.io/guides/aws-lambda

```
Caused by: java.lang.IllegalStateException: Multiple extensions registered a feature of the same name: amazon-lambda
	at io.quarkus.deployment.steps.MainClassBuildStep.build(MainClassBuildStep.java:297)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
```

> I have the need to write a lambda to invoke another lambda, so I though that using the `io.quarkiverse.amazonservices:quarkus-amazon-services` LambdaClient would be the correct approach.
> 
> On 2.8.1 I get the error mentioned in this issue
> 
> ```
> java.lang.RuntimeException: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
> 	[error]: Build step io.quarkus.amazon.lambda.deployment.LambdaProcessor#discoverTelemetry threw an exception: java.lang.IncompatibleClassChangeError: Found class io.quarkus.amazon.lambda.runtime.LambdaBuildTimeConfig, but interface was expected
> 	at io.quarkus.amazon.lambda.deployment.LambdaProcessor.discoverTelemetry(LambdaProcessor.java:99)
> 	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
> 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
> ```
> 
> On 3.0.0.alpha1 the error below since I'm using https://quarkus.io/guides/aws-lambda
> 
> ```
> Caused by: java.lang.IllegalStateException: Multiple extensions registered a feature of the same name: amazon-lambda
> 	at io.quarkus.deployment.steps.MainClassBuildStep.build(MainClassBuildStep.java:297)
> 	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:733)
> 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:856)
> 	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
> ```

too bad, I missed you comment after releasing. Will do a 3.0.1 with another feature name.

--- a/apigatewaymanagementapi/deployment/src/main/java/io/quarkiverse/amazon/apigatewaymanagementapi/deployment/ApiGatewayManagementApiProcessor.java
+++ b/apigatewaymanagementapi/deployment/src/main/java/io/quarkiverse/amazon/apigatewaymanagementapi/deployment/ApiGatewayManagementApiProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.apigatewaymanagementapi.ApiGatewayManagem
 
 public class ApiGatewayManagementApiProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-apigatewaymanagementapi";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-apigatewaymanagementapi";
 
     ApiGatewayManagementApiBuildTimeConfig buildTimeConfig;
 

--- a/cloudwatch/deployment/src/main/java/io/quarkiverse/amazon/cloudwatch/deployment/CloudWatchProcessor.java
+++ b/cloudwatch/deployment/src/main/java/io/quarkiverse/amazon/cloudwatch/deployment/CloudWatchProcessor.java
@@ -26,7 +26,7 @@ import software.amazon.awssdk.services.cloudwatch.CloudWatchClientBuilder;
 
 public class CloudWatchProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-cloudwatch";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-cloudwatch";
 
     CloudWatchBuildTimeConfig buildTimeConfig;
 

--- a/cloudwatchlogs/deployment/src/main/java/io/quarkiverse/amazon/cloudwatch/deployment/CloudWatchLogsProcessor.java
+++ b/cloudwatchlogs/deployment/src/main/java/io/quarkiverse/amazon/cloudwatch/deployment/CloudWatchLogsProcessor.java
@@ -26,7 +26,7 @@ import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClientBuilde
 
 public class CloudWatchLogsProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-cloudwatchlogs";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-cloudwatchlogs";
 
     CloudWatchLogsBuildTimeConfig buildTimeConfig;
 

--- a/cognito-user-pools/deployment/src/main/java/io/quarkiverse/amazon/cognitouserpools/deployment/CognitoUserPoolsProcessor.java
+++ b/cognito-user-pools/deployment/src/main/java/io/quarkiverse/amazon/cognitouserpools/deployment/CognitoUserPoolsProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.cognitoidentityprovider.CognitoIdentityPr
 
 public class CognitoUserPoolsProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-cognito-user-pools";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-cognito-user-pools";
 
     CognitoUserPoolsBuildTimeConfig buildTimeConfig;
 

--- a/dynamodb-enhanced/deployment/src/main/java/io/quarkiverse/amazon/dynamodb/enhanced/deployment/DynamodbEnhancedProcessor.java
+++ b/dynamodb-enhanced/deployment/src/main/java/io/quarkiverse/amazon/dynamodb/enhanced/deployment/DynamodbEnhancedProcessor.java
@@ -61,7 +61,7 @@ import software.amazon.awssdk.services.dynamodb.DynamoDbAsyncClient;
 import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
 
 public class DynamodbEnhancedProcessor {
-    private static final String FEATURE = "amazon-dynamodb-enhanced";
+    private static final String FEATURE = "amazon-sdk-dynamodb-enhanced";
 
     @BuildStep
     FeatureBuildItem feature() {

--- a/dynamodb/deployment/src/main/java/io/quarkiverse/amazon/dynamodb/deployment/DynamodbProcessor.java
+++ b/dynamodb/deployment/src/main/java/io/quarkiverse/amazon/dynamodb/deployment/DynamodbProcessor.java
@@ -42,7 +42,7 @@ import software.amazon.awssdk.services.dynamodb.DynamoDbClientBuilder;
 
 public class DynamodbProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-dynamodb";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-dynamodb";
 
     DynamodbBuildTimeConfig buildTimeConfig;
 

--- a/ecr/deployment/src/main/java/io/quarkiverse/amazon/ecr/deployment/EcrProcessor.java
+++ b/ecr/deployment/src/main/java/io/quarkiverse/amazon/ecr/deployment/EcrProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.ecr.EcrClientBuilder;
 
 public class EcrProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-ecr";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-ecr";
 
     EcrBuildTimeConfig buildTimeConfig;
 

--- a/eventbridge/deployment/src/main/java/io/quarkiverse/amazon/eventbridge/deployment/EventBridgeProcessor.java
+++ b/eventbridge/deployment/src/main/java/io/quarkiverse/amazon/eventbridge/deployment/EventBridgeProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.eventbridge.EventBridgeClientBuilder;
 
 public class EventBridgeProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-eventbridge";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-eventbridge";
 
     EventBridgeBuildTimeConfig buildTimeConfig;
 

--- a/iam/deployment/src/main/java/io/quarkiverse/amazon/iam/deployment/IamProcessor.java
+++ b/iam/deployment/src/main/java/io/quarkiverse/amazon/iam/deployment/IamProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.iam.IamClientBuilder;
 
 public class IamProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-iam";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-iam";
 
     IamBuildTimeConfig buildTimeConfig;
 

--- a/inspector/deployment/src/main/java/io/quarkiverse/amazon/inspector/deployment/InspectorProcessor.java
+++ b/inspector/deployment/src/main/java/io/quarkiverse/amazon/inspector/deployment/InspectorProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.inspector.InspectorClientBuilder;
 
 public class InspectorProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-inspector";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-inspector";
 
     InspectorBuildTimeConfig buildTimeConfig;
 

--- a/inspector2/deployment/src/main/java/io/quarkiverse/amazon/inspector2/deployment/Inspector2Processor.java
+++ b/inspector2/deployment/src/main/java/io/quarkiverse/amazon/inspector2/deployment/Inspector2Processor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.inspector2.Inspector2ClientBuilder;
 
 public class Inspector2Processor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-inspector2";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-inspector2";
 
     Inspector2BuildTimeConfig buildTimeConfig;
 

--- a/kinesis/deployment/src/main/java/io/quarkiverse/amazon/kinesis/deployment/KinesisProcessor.java
+++ b/kinesis/deployment/src/main/java/io/quarkiverse/amazon/kinesis/deployment/KinesisProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.kinesis.KinesisClientBuilder;
 
 public class KinesisProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-kinesis";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-kinesis";
 
     KinesisBuildTimeConfig buildTimeConfig;
 

--- a/kms/deployment/src/main/java/io/quarkiverse/amazon/kms/deployment/KmsProcessor.java
+++ b/kms/deployment/src/main/java/io/quarkiverse/amazon/kms/deployment/KmsProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.kms.KmsClientBuilder;
 
 public class KmsProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-kms";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-kms";
 
     KmsBuildTimeConfig buildTimeConfig;
 

--- a/lambda/deployment/src/main/java/io/quarkiverse/amazon/lambda/deployment/LambdaProcessor.java
+++ b/lambda/deployment/src/main/java/io/quarkiverse/amazon/lambda/deployment/LambdaProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.lambda.LambdaClient;
 import software.amazon.awssdk.services.lambda.LambdaClientBuilder;
 
 public class LambdaProcessor extends AbstractAmazonServiceProcessor {
-    private static final String AMAZON_CLIENT_NAME = "amazon-lambda";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-lambda";
 
     LambdaBuildTimeConfig buildTimeConfig;
 

--- a/paymentcryptography/deployment/src/main/java/io/quarkiverse/amazon/paymentcryptography/deployment/PaymentCryptographyProcessor.java
+++ b/paymentcryptography/deployment/src/main/java/io/quarkiverse/amazon/paymentcryptography/deployment/PaymentCryptographyProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.paymentcryptography.PaymentCryptographyCl
 
 public class PaymentCryptographyProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-paymentcryptography";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-paymentcryptography";
 
     PaymentCryptographyBuildTimeConfig buildTimeConfig;
 

--- a/paymentcryptographydata/deployment/src/main/java/io/quarkiverse/amazon/paymentcryptographydata/deployment/PaymentCryptographyDataProcessor.java
+++ b/paymentcryptographydata/deployment/src/main/java/io/quarkiverse/amazon/paymentcryptographydata/deployment/PaymentCryptographyDataProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.paymentcryptographydata.PaymentCryptograp
 
 public class PaymentCryptographyDataProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-paymentcryptographydata";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-paymentcryptographydata";
 
     PaymentCryptographyDataBuildTimeConfig buildTimeConfig;
 

--- a/s3-transfer-manager/deployment/src/main/java/io/quarkiverse/amazon/s3/deployment/S3TransferManagerProcessor.java
+++ b/s3-transfer-manager/deployment/src/main/java/io/quarkiverse/amazon/s3/deployment/S3TransferManagerProcessor.java
@@ -36,7 +36,7 @@ public class S3TransferManagerProcessor {
             BuildProducer<FeatureBuildItem> feature,
             BuildProducer<AmazonClientInterceptorsPathBuildItem> interceptors) {
 
-        feature.produce(new FeatureBuildItem("amazon-s3-transfer-manager"));
+        feature.produce(new FeatureBuildItem("amazon-sdk-s3-transfer-manager"));
     }
 
     @BuildStep

--- a/s3/deployment/src/main/java/io/quarkiverse/amazon/s3/deployment/S3Processor.java
+++ b/s3/deployment/src/main/java/io/quarkiverse/amazon/s3/deployment/S3Processor.java
@@ -43,7 +43,7 @@ import software.amazon.awssdk.services.s3.presigner.S3Presigner;
 
 public class S3Processor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-s3";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-s3";
 
     S3BuildTimeConfig buildTimeConfig;
 

--- a/secretsmanager/deployment/src/main/java/io/quarkiverse/amazon/secretsmanager/deployment/SecretsManagerProcessor.java
+++ b/secretsmanager/deployment/src/main/java/io/quarkiverse/amazon/secretsmanager/deployment/SecretsManagerProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.secretsmanager.SecretsManagerClientBuilde
 
 public class SecretsManagerProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-secretsmanager";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-secretsmanager";
 
     SecretsManagerBuildTimeConfig buildTimeConfig;
 

--- a/ses/deployment/src/main/java/io/quarkiverse/amazon/ses/deployment/SesProcessor.java
+++ b/ses/deployment/src/main/java/io/quarkiverse/amazon/ses/deployment/SesProcessor.java
@@ -42,7 +42,7 @@ import software.amazon.awssdk.services.ses.SesClientBuilder;
 
 public class SesProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-ses";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-ses";
 
     SesBuildTimeConfig buildTimeConfig;
 

--- a/sfn/deployment/src/main/java/io/quarkiverse/amazon/sfn/deployment/SfnProcessor.java
+++ b/sfn/deployment/src/main/java/io/quarkiverse/amazon/sfn/deployment/SfnProcessor.java
@@ -42,7 +42,7 @@ import software.amazon.awssdk.services.sfn.SfnClientBuilder;
 
 public class SfnProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-sfn";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-sfn";
 
     SfnBuildTimeConfig buildTimeConfig;
 

--- a/sns/deployment/src/main/java/io/quarkiverse/amazon/sns/deployment/SnsProcessor.java
+++ b/sns/deployment/src/main/java/io/quarkiverse/amazon/sns/deployment/SnsProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.sns.SnsClientBuilder;
 
 public class SnsProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-sns";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-sns";
 
     SnsBuildTimeConfig buildTimeConfig;
 

--- a/sqs/deployment/src/main/java/io/quarkiverse/amazon/sqs/deployment/SqsProcessor.java
+++ b/sqs/deployment/src/main/java/io/quarkiverse/amazon/sqs/deployment/SqsProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.sqs.SqsClientBuilder;
 
 public class SqsProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-sqs";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-sqs";
 
     SqsBuildTimeConfig buildTimeConfig;
 

--- a/ssm/deployment/src/main/java/io/quarkiverse/amazon/ssm/deployment/SsmProcessor.java
+++ b/ssm/deployment/src/main/java/io/quarkiverse/amazon/ssm/deployment/SsmProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.ssm.SsmClientBuilder;
 
 public class SsmProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-ssm";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-ssm";
 
     SsmBuildTimeConfig buildTimeConfig;
 

--- a/sts/deployment/src/main/java/io/quarkiverse/amazon/sts/deployment/StsProcessor.java
+++ b/sts/deployment/src/main/java/io/quarkiverse/amazon/sts/deployment/StsProcessor.java
@@ -41,7 +41,7 @@ import software.amazon.awssdk.services.sts.StsClientBuilder;
 
 public class StsProcessor extends AbstractAmazonServiceProcessor {
 
-    private static final String AMAZON_CLIENT_NAME = "amazon-sts";
+    private static final String AMAZON_CLIENT_NAME = "amazon-sdk-sts";
 
     StsBuildTimeConfig buildTimeConfig;
 

